entry_types/scrolled/package/spec/frontend/usePhonePlatform/inEditorPreview-spec.js

```diff
@@ -0,0 +1,36 @@
+
+import {PhonePlatformProvider} from 'frontend';
+import {usePhonePlatform} from 'frontend/usePhonePlatform';
+import {loadInlineEditingComponents} from 'frontend/inlineEditing';
+
+import {renderHook, act} from '@testing-library/react-hooks';
+
+import '@testing-library/jest-dom/extend-expect'
+import {tick} from "support";
+
+
+describe('usePhonePlatform', () => {
+  beforeAll(loadInlineEditingComponents);
+
+  it('sets value when emulation mode changes', async () => {
+    const {result} = renderHook(() => usePhonePlatform(), {wrapper: PhonePlatformProvider});
+
+    await asyncHandlingOfMessage(() => {
+      window.postMessage({
+        type: 'CHANGE_EMULATION_MODE',
+        payload: 'phone'
+      }, '*');
+    });
+
+    expect(result.current).toEqual(true);
+  });
+  
+  async function asyncHandlingOfMessage(callback) {
+    await act(async () => {
+      callback();
+      // Ensure React update triggered by async handling of message is
+      // wrapped in act
+      await tick();
+    });
+  }
+});
```

Comment:
**suggestion**: Extract test helper

As already suggested [here](https://redmine.codevise.de/issues/17670#note-23) by @aviav, now that we have the third place in the code base that uses this "await-act-tick-trick" (here are the [other](https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/spec/frontend/features/scrollToSectionMessage-spec.js#L26) [two](https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/spec/contentElements/dataWrapperChart/useIframeHeight-spec.js#L48)), it might be time to extract the helper.

When we move the function to `spec/support/`, I'd suggest to rename it to `asyncHandlingOf` since the helper itself does not really care about what is handled asynchronously. Together with the `await` and the actual callback, the whole line will read as a sentence that says asynchronous handling of what is being awaited.

Reply:
While replacing the helper [here](https://github.com/codevise/pageflow/blob/master/entry_types/scrolled/package/spec/frontend/features/scrollToSectionMessage-spec.js#L26), it doesn't work for some reason. I have tried a lot.
Any idea?

Reply:
The problem is that `render` from `@testing-library/react` and  `renderHook` from `@testing-library/react` each use a different implementation of `act`. We therefore cannot have a shared `asyncHandlingOf` helper that works for both hook and component test.

For hooks, I found that there is also [`waitForNextUpdate`](https://react-hooks-testing-library.com/usage/advanced-hooks#async) which solves a similar problem. But this does not help in the case of `useIframeHeight` when we want to assert that no change happens. There is no way to "wait for nothing to happen".

My best idea so far therefore is to have two copies of the helper: one in `spec/support/asyncHandlingOf/index.js` that uses `act` form `@testing-library/react`, the other in `spec/support/asyncHandlingOf/forHooks.js` that uses `act` from `@testing-library/react-hooks`.